src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -753,81 +753,107 @@ private void ProcessApp14Marker(BufferedReadStream stream, int remaining)
                             table[j] = this.temp[j];
                         }
+
+                        break;
                     }
 
-                    break;
+                    // 16 bit values
                     case 1:
                     {
-                        // 16 bit values
+                        // Validate: 16 bit table needs exactly 128 bytes
                         if (remaining < 128)
                         {
-                            done = true;
-                            break;
+                            JpegThrowHelper.ThrowBadMarker(nameof(JpegConstants.Markers.DQT), remaining);
                         }
 
                         stream.Read(this.temp, 0, 128);
                         remaining -= 128;
 
-                        ref Block8x8F table = ref this.QuantizationTables[tableIndex];
                         for (int j = 0; j < 64; j++)
                         {
                             table[j] = (this.temp[2 * j] << 8) | this.temp[(2 * j) + 1];
                         }
-                    }
 
-                    break;
+                        break;
+                    }
 
+                    // Unknown precision - error
                     default:
                     {
-                        JpegThrowHelper.ThrowBadQuantizationTable();
+                        JpegThrowHelper.ThrowBadQuantizationTablePrecision(tablePrecision);
                         break;
                     }
                 }
 
-                if (done)
+                // Estimating quality
+                switch (tableIndex)
                 {
-                    break;
-                }
-            }
+                    // luminance table
+                    case 0:
+                    {
+                        Quantization.EstimateQuality(ref table, Quantization.UnscaledQuant_Luminance, out double quality, out double variance);
+                        jpegMetadata.LumaQuality = quality;
+                        if (variance <= Quantization.StandardLuminanceTableVarianceThreshold)
+                        {
+                            jpegMetadata.LumaQuantizationTable = table.RoundAsInt16Block();
+                        }
 
-            if (remaining != 0)
-            {
-                JpegThrowHelper.ThrowBadMarker(nameof(JpegConstants.Markers.DQT), remaining);
-            }
+                        break;
+                    }
 
-            this.Metadata.GetFormatMetadata(JpegFormat.Instance).Quality = QualityEvaluator.EstimateQuality(this.QuantizationTables);
+                    // chrominance table
+                    case 1:
+                    {
+                        Quantization.EstimateQuality(ref table, Quantization.UnscaledQuant_Chrominance, out double quality, out double variance);
+                        jpegMetadata.ChromaQuality = quality;
+                        if (variance <= Quantization.StandardChrominanceTableVarianceThreshold)
```

Comment:
Is it common that luma table is non-standard while chroma table is standard, or vice-versa?

Reply:
It can be within some variance threshold but not that I'm aware of.